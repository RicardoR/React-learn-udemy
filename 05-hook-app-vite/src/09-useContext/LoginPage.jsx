import {useContext} from "react";
import {UserContext} from "./context/UserContext";

export const LoginPage = () => {

    const {user, setUser} = useContext(UserContext);

    const loginUser = () => {
        const user = {
            id: 123,
            name: 'Ricardo',
            mail: 'ricardo@mail'
        }
        setUser(user);
    }

    return (
        <>
            <h1> LoginPage</h1>
            <hr/>
            <pre>
                {JSON.stringify(user, null, 3)}
            </pre>
            <button
                onClick={loginUser}
                className='btn btn-primary'>
                Set user
            </button>
        </>
    );
};

