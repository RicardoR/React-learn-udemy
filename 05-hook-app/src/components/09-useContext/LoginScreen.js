import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleLogin = () => {
    setUser({
      id: 123,
      name: 'John Doe',
    });
  };

  return (
    <div>
      <h1>LoginScreen</h1>
      <hr />
      <pre>{JSON.stringify(user, null, 3)}</pre>

      <button className="btn btn-primary" onClick={() => handleLogin()}>
        Login
      </button>
    </div>
  );
};
