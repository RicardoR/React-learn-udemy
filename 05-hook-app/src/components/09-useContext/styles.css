.navbar {
  padding: 0.5rem;
  margin-bottom: 24px;
}
