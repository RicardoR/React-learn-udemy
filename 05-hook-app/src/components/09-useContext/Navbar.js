import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './styles.css';

export const Navbar = () => {
  return (
    <nav className="navbar navbar-expand-md navbar-dark bg-dark">
      <Link className="navbar-brand" to="#">
        useContext
      </Link>

      <button
        class="navbar-toggler"
        type="button"
        data-toggle="collapse"
        data-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          <li className="nav-item">
            <NavLink className="nav-link" to="/">
              Home
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/about">
              About
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink className="nav-link" to="/login">
              Login
            </NavLink>
          </li>
        </ul>
      </div>
    </nav>
  );
};
