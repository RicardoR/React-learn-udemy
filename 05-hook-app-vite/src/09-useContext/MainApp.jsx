import {Navigate, Route, Routes} from "react-router-dom";
import {HomePage} from "./HomePage";
import {AboutPage} from "./AboutPage";
import {LoginPage} from "./LoginPage";
import {Navbar} from "./Navbar";

export const MainApp = () => {
    return (
        <>
            <Navbar/>
            <Routes>
                <Route path='/' element={<HomePage/>}/>
                <Route path='about' element={<AboutPage/>}/>
                <Route path='login' element={<LoginPage/>}/>
                <Route path='/*' element={<Navigate to='login'/>}/>
            </Routes>
        </>
    );
};

