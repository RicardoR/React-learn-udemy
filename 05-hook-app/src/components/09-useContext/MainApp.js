import React from 'react';
import { Outlet } from 'react-router-dom';
import { Navbar } from './Navbar';
import { UserContext } from './UserContext';

export const MainApp = () => {
  const user = {
    id: 123,
    name: 'Ricardo',
    email: 'email@email.com',
  };
  return (
    <div>
      <UserContext.Provider value={user}>
        <Navbar />
        {/* An <Outlet> renders whatever child route is currently active,
          so you can think about this <Outlet> as a placeholder for
          the child routes we defined above. */}
        <Outlet />
      </UserContext.Provider>
    </div>
  );
};
