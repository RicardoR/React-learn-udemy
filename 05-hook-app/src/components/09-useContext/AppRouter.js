import React from 'react';
import { Routes, Route } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { MainApp } from './MainApp';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
  return (
    <div>
      <Routes>
        <Route path="/" element={<MainApp />}>
          <Route index element={<HomeScreen />} />
          <Route path="about" element={<AboutScreen />} />
          <Route path="login" element={<LoginScreen />} />

          {/* Using path="*"" means "match anything", so this route
                acts like a catch-all for URLs that we don't have explicit
                routes for. */}
          <Route path="*" element={<HomeScreen />} />
        </Route>
      </Routes>
    </div>
  );
};
