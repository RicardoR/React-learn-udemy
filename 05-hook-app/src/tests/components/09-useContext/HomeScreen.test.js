import React from 'react';
import { mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('HomeScreen UT', () => {
  const user = {
    name: 'Jane Doe',
    email: 'jan@mail.com',
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user }}>
      <HomeScreen />
    </UserContext.Provider>
  );

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should display user name', () => {
    expect(wrapper.find('h1').text()).toBe('HomeScreen');
    expect(wrapper.find('pre').text()).toBe(JSON.stringify(user, null, 3));
  });
});
