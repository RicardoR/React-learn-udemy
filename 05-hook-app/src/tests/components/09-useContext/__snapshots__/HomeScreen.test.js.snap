// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen UT should display correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Jane Doe",
   "email": "jan@mail.com"
}
  </pre>
</div>
`;
